### PR TITLE
Login functionality implemented separately

### PR DESCRIPTION
**Pull request by menynexi**
I have implemented the Login functionality separately to make it easier to code review and to test out any bugs that may occur prior to implementation 

Please use the following link to test the login function 
https://mangutierrez-step-20202.uc.r.appspot.com/login.html

I have also resolved the issue with the maxDisplay. NowIf there are not enough comments to display the users input it will display all comments. Additionally I have linked a document underlining my troubleshooting process.

https://docs.google.com/document/d/17NOg7Xvy7e7cUNa_eTDE3vbt_WOaP_v91u1McBSjahI/edit?usp=sharing

### CODE REVIEW

**Review by WillHipschman — CHANGES_REQUESTED** [review]
Can you split this into two pull requests? One for the login functionality and one for fixing max comments.

**Comment by DirtGrubDylan** [comment]
So taking a look, how do I know that I am logged into your page? I didn't see any indication unless I logged in and then went back to the login page. Is this something you can persist across all html pages in future PRs?

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 1]
```diff
@@ -57,13 +58,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       setMaxComment(request, request.getParameter(MAXCOMMENT_PARAMETER));
     }
 
-    response.sendRedirect(COMMENT_HTML_PARAMETER);
+    response.sendRedirect(this.COMMENT_HTML_PARAMETER);
```
Comment (DirtGrubDylan) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:61]: So while `this` can be used here, I think we need to be consistent (i.e. use `this` for all member variable access or never use it)

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType(TEXT_TO_HTML);
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String logoutUrl = userService.createLogoutURL(URL_TO_REDIRECT_AFTER_USER_LOGSOUT);
+
+      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
```
Comment (DirtGrubDylan) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:42]: Bust these hard coded strings out

**Review by DirtGrubDylan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**